### PR TITLE
configure --with-default-mail-command-path: specify default mail command

### PR DESCRIPTION
**Pull request by schaal**
Add option to configure to specify the default mail command, if not given falls back to config.h as of now

### CODE REVIEW

File: logrotate.8.in [hunk 1]
```diff
@@ -58,8 +58,8 @@ output logged to that file is the same as when running \fBlogrotate\fR with
 Tells \fBlogrotate\fR which command to use when mailing logs. This
 command should accept two arguments: 1) the subject of the message, and
 2) the recipient. The command must then read a message on standard input
-and mail it to the recipient. The default mail command is \fB/bin/mail
--s\fR.
+and mail it to the recipient. The default mail command is shown in the
+output from \fBlogrotate --help\fR.
```
Comment (glensc) [inline logrotate.8.in]: clever. however this is `.in` file, so `@DEFAULT_MAIL_COMMAND@` should be available as autoconf substitution.

File: configure.ac [hunk 2]
```diff
@@ -64,6 +64,22 @@ AC_ARG_WITH([state-file-path],
 AC_DEFINE_UNQUOTED([STATEFILE], "$STATE_FILE_PATH")
 AC_SUBST(STATE_FILE_PATH)
 
+AC_ARG_WITH([default-mail-command-path],
+  AC_HELP_STRING([--with-default-mail-command-path=PATH],
+                 [path to default mail command (based on config.h if not given)]),
```
Comment (glensc) [inline configure.ac]: it is not "path", but "command", i.e the default value is "path" + "arguments": `/bin/mail -s`, so `--with-default-mail-command` would be more appropriate.

File: logrotate.8.in (hunk 1 above)
Comment (schaal) [inline logrotate.8.in]: I tried to keep the current definitions from config.h if configure is called without this option, but then @DEFAULT_MAIL_COMMAND@ is not available in autoconf

File: configure.ac (hunk 2 above)
Comment (schaal) [inline configure.ac]: Right, makes more sense, will update with your suggestion

File: logrotate.8.in (hunk 1 above)
Comment (glensc) [inline logrotate.8.in]: ok then by me. ideally the `config.h` platform defaults should be moved to configure or dropped at all.

**Comment by kdudka** [comment]
On Monday, January 02, 2017 08:22:49 Elan Ruusam�e wrote:
> ideally the `config.h` platform defaults should be moved to
> configure or dropped at all.

Exactly.  Do you feel like writing such a patch?

As for this PR, I am fine with merging it as it is.

Thank you all for working on this!

File: configure.ac [hunk 3]
```diff
@@ -64,6 +64,22 @@ AC_ARG_WITH([state-file-path],
 AC_DEFINE_UNQUOTED([STATEFILE], "$STATE_FILE_PATH")
 AC_SUBST(STATE_FILE_PATH)
 
+AC_ARG_WITH([default-mail-command],
+  AC_HELP_STRING([--with-default-mail-command=COMMAND],
+                 [default mail command (based on config.h if not given)]),
+  [
+    case "$withval" in
+      yes|no)
+        AC_MSG_ERROR([--with-default-mail-command-path=COMMAND requires a command to be specified])
```
Comment (glensc) [inline configure.ac]: please fix here too

**Pull request closed without merging**